Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.!

In [1]:
import pandas as pd
from scipy import stats
from scipy.stats import norm

In [2]:
data = pd.read_csv("/content/BuyerRatio (1).csv")

In [3]:
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [4]:
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Assume, Null hypothesis (Ho): Independence of categorical variables(males-females buyer ration are similar across regions). Alternate Hypothesis (Ha):Dependence of categorical variables(males-females buyer ration are not similar across regions).



In [5]:
file = data.iloc[:,1:6]
file

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [6]:
file.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [8]:
stats.chi2_contingency(file.values) 

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

As we can see , p_val = 0.6603 > 0.05. therefore we accept Null hypothesis, that males-females buyer ration are similar across regions